flash/vision/style_transfer/model.py

```diff
@@ -0,0 +1,227 @@
+
+    def forward(self, input):
+        return input / 255.0
+
+
+class Transformer(nn.Module):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.encoder = nn.Sequential(
+            Conv(3, 32, kernel_size=9),
+            Conv(32, 64, kernel_size=3, stride=2),
+            Conv(64, 128, kernel_size=3, stride=2),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+        )
+        self.decoder = nn.Sequential(
+            Conv(128, 64, kernel_size=3, stride=2, upsample=True),
+            Conv(64, 32, kernel_size=3, stride=2, upsample=True),
+            Conv(32, 3, kernel_size=9, norm=False, activation=False),
+        )
+
+        self.preprocessor = FloatToUint8Range()
+        self.postprocessor = Uint8ToFloatRange()
+
+    def forward(self, input):
+        input = self.preprocessor(input)
+        output = self.decoder(self.encoder(input))
+        return self.postprocessor(output)
+
+
+class StyleTransfer(Task):
+
+    def __init__(
+        self,
+        style_image: Union[str, torch.Tensor],
+        model: Optional[nn.Module] = None,
+        multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None,
+        content_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        style_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        learning_rate: float = 1e-3,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+    ):
+        self.save_hyperparameters(ignore="style_image")
+
+        if isinstance(style_image, str):
+            style_image = read_image(style_image)
+
+        if model is None:
+            # TODO: import this from pystiche
+            model = Transformer()
+
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+
+        if content_loss is None:
+            content_loss = self.default_content_loss(multi_layer_encoder)
+
+        if style_loss is None:
+            style_loss = self.default_style_loss(multi_layer_encoder)
+
+        perceptual_loss = loss.PerceptualLoss(content_loss, style_loss)
+
+        super().__init__(
+            model=model,
+            loss_fn=perceptual_loss,
+            optimizer=optimizer,
+            optimizer_kwargs=optimizer_kwargs,
+            scheduler=scheduler,
+            scheduler_kwargs=scheduler_kwargs,
+            learning_rate=learning_rate,
+            serializer=serializer,
+        )
+
+        self.perceptual_loss = perceptual_loss
+        self.perceptual_loss.set_style_image(style_image)
```

Comment:
Is it possible to set multiple style_images and to randomly sample from them ?

Reply:
There are approaches like that, but they go beyond a simple entry example.